applications/serial_lte_modem/src/slm_at_cmng.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <logging/log.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <string.h>
+#include <net/socket.h>
+#include <modem/modem_info.h>
+#include <modem/modem_key_mgmt.h>
+#include "slm_util.h"
+#include "slm_at_host.h"
+#include "slm_native_tls.h"
+#include "slm_at_cmng.h"
+
+LOG_MODULE_REGISTER(cmng, CONFIG_SLM_LOG_LEVEL);
+
+/**@brief List of supported AT commands. */
+enum slm_cmng_at_cmd {
+	AT_XCMNG,
+	AT_CMNG_MAX
+};
+
+/**@brief List of supported opcode */
+enum slm_cmng_opcode {
+	AT_CMNG_OP_WRITE,
+	AT_CMNG_OP_LIST,
+	AT_CMNG_OP_READ,
+	AT_CMNG_OP_DELETE
+};
+
+/**@brief List of supported type */
+enum slm_cmng_type {
+	AT_CMNG_TYPE_CA_CERT,
+	AT_CMNG_TYPE_CERT,
+	AT_CMNG_TYPE_PRIV,
+	AT_CMNG_TYPE_PSK,
+	AT_CMNG_TYPE_PSK_ID,
+};
+
+/* global functions defined in different files */
+void rsp_send(const uint8_t *str, size_t len);
+
+/** forward declaration of cmd handlers **/
+static int handle_at_xcmng(enum at_cmd_type cmd_type);
+
+/**@brief SLM AT Command list type. */
+static slm_at_cmd_list_t cmng_at_list[AT_CMNG_MAX] = {
+	{AT_XCMNG, "AT%CMNG", handle_at_xcmng},
+};
+
+/* global variable defined in different files */
+extern struct at_param_list at_param_list;
+extern char rsp_buf[CONFIG_AT_CMD_RESPONSE_MAX_LEN];
+
+/**@brief handle AT#XCMNG commands
+ *  AT#XCMNG=<opcode>[,<sec_tag>[,<type>[,<content>]]]
+ *  AT#XCMNG? READ command not supported
+ *  AT#XCMNG=? READ command not supported
+ */
+static int handle_at_xcmng(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op, type;
+	nrf_sec_tag_t sec_tag;
+	uint8_t *content;
+	size_t len = CONFIG_AT_CMD_RESPONSE_MAX_LEN;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		if (at_params_valid_count_get(&at_param_list) < 2) {
+			return -EINVAL;
+		}
+		err = at_params_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op > AT_CMNG_OP_DELETE) {
+			return -EINVAL;
+		}
+		if (op == AT_CMNG_OP_LIST) {
+			/* Currently not support list command */
+			return -EINVAL;
+		}
+		if (at_params_valid_count_get(&at_param_list) < 4) {
+			/* READ, WRITE, DELETE requires sec_tag and type */
+			return -EINVAL;
+		}
+		err = at_params_int_get(&at_param_list, 2, &sec_tag);
+		if (err < 0) {
+			return err;
+		};
+		if (sec_tag > MAX_SLM_SEC_TAG) {
+			return -EINVAL;
+		}
+		err = at_params_short_get(&at_param_list, 3, &type);
+		if (err < 0) {
+			return err;
+		};
+		if (op == AT_CMNG_OP_WRITE) {
+			content = k_malloc(CONFIG_AT_CMD_RESPONSE_MAX_LEN);
+			err = at_params_string_get(&at_param_list, 4, content,
```

Comment:
Need to check param count before get since this is mandatory param

Reply:
Fixed.